velox/core/SimpleFunctionMetadata.h

```diff
@@ -321,12 +375,38 @@ class SimpleFunctionMetadata : public ISimpleFunctionMetadata {
   explicit SimpleFunctionMetadata(std::shared_ptr<const Type> returnType)
       : returnType_(
             returnType ? std::move(returnType) : CppToType<TReturn>::create()) {
+    setSignatureAndPriority();
     verifyReturnTypeCompatibility();
   }
 
   ~SimpleFunctionMetadata() override = default;
 
-  std::shared_ptr<exec::FunctionSignature> signature() const final {
+  const std::shared_ptr<exec::FunctionSignature> signature() const override {
+    return signatrue_;
+  }
+
+  std::string helpMessage(const std::string& name) const final {
+    std::string s{name};
+    s.append("(");
+    bool first = true;
+    for (auto& arg : argTypes()) {
+      if (!first) {
+        s.append(", ");
+      }
+      first = false;
+      s.append(arg->toString());
+    }
+
+    if (isVariadic()) {
+      s.append("...");
+    }
+
+    s.append(")");
+    return s;
+  }
+
+ private:
+  void setSignatureAndPriority() {
```

Comment:
This is will make extending use cases for TypeAnalysisResults difficult, e.g. if we want to compute some new variable X based on it, would we rename this function setSignatureAndPriorityAndX()?

What about making TypeAnalysisResults a private member variable, so that we could use it outside this function to initialize priority.

For signature, the main reason the building of TypeAnalysisResults and the signature are coupled is because we clear out the type strings for the return type and the arguments.

One way to address this would be to make TypeAnalysisResults templated on TReturn and Args... and in the constructor we could initialize the string value for return and a vector of strings for args.  If we update the run functions to take results and an std::ostringstream we could make out a temporary instead of a field in the object and that might be cleaner too.

Since the TypeAnalysisResults knows whether or not the args have variadic, we could even make the function to generate the FunctionSignature part of TypeAnalysisResults then too.

Then we could get rid of this function and make all these variables const in this class.  Also, SimpleFunctionMetadata wouldn't need to know about the TypeAnalysis structs which would be nicer too (only TypeAnalysisResults itself is exposed).